### PR TITLE
Add `word_document_generator` module

### PR DESCRIPTION
**Pull request by MikeEvansLarah**
The module includes the `WordDocumentGenerator` class which can be used for rendering Word `.docx` templates.

### CODE REVIEW

File: src/corvus_python/word_document_generator/word_document_generator.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            context (Dict[str, any]): The context to render the template with.
+
+        Returns:
+            bytes: The rendered document as a byte array.
+        """
+
+        template.render(context)
+
+        # save to byte array
+        stream = BytesIO()
```
Comment (jongeorge1) [inline src/corvus_python/word_document_generator/word_document_generator.py:27]: if you replace this line with
`with BytesIO() as stream:`

then indent the next couple of lines, you can lose the explicit call to `stream.close()` as well as having the same effect as using a try...finally block.

File: src/corvus_python/word_document_generator/word_document_generator.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        template.save(stream)
+        result = stream.getvalue()
+
+        # dispose of the stream
+        stream.close()
```
Comment (jongeorge1) [inline src/corvus_python/word_document_generator/word_document_generator.py:32]: I think it's good practice to put this `stream.close()` in a `finally` block to ensure it gets called.

File: tests/bdd/word_document_generator/generate_document.feature [hunk 3]
```diff
@@ -0,0 +1,41 @@
+@word_document_generator
+Feature: Generate Word document from template
+
+    Scenario: Generate Word document from template
+        Given I have a template file 'template1.docx'
+        And I read the template context from 'context1.json'
+        When I generate a Word document named 'output1.docx'
+        Then the Word document should be generated
+        # We should also check the content of the generated document but it's not easy to do
```
Comment (jongeorge1) [inline tests/bdd/word_document_generator/generate_document.feature:9]: Is the rendering process deterministic?

If so, we could run this once, manually verify the output and then store it alongside the input and output folders. Then we could validate the test output by loading the expected output and comparing byte by byte.

File: tests/bdd/word_document_generator/input/image1.jpg
Comment (jongeorge1) [inline tests/bdd/word_document_generator/input/image1.jpg:1]: This was not what I expected to see in the PR.

File: tests/bdd/word_document_generator/generate_document.feature (hunk 3 above)
Comment (MikeEvansLarah) [reply tests/bdd/word_document_generator/generate_document.feature:9]: I did wonder that, but I didn't know if there was any metadata stored in the file that might throw the comparison off. I can try it out.

Comment (MikeEvansLarah) [reply tests/bdd/word_document_generator/generate_document.feature:9]: It seems it's not deterministic, presumably due to metadata

**Review by jongeorge1 — APPROVED** [review]

**Pull request merged**